packages/@aws-cdk/aws-pipes-targets-alpha/test/lambda.test.ts

```diff
@@ -0,0 +1,184 @@
+        LambdaFunctionParameters: {
+          InvocationType: 'REQUEST_RESPONSE',
+        },
+      },
+    });
+  });
+
+  it('should have input transformation', () => {
+    // ARRANGE
+    const app = new App();
+    const stack = new Stack(app, 'TestLambdaFunctionStack');
+
+    const lambdaFunction = new lambda.Function(stack, 'MyLambda', {
+      code: lambda.Code.fromInline('console.log("Hello, world!")'),
+      runtime: lambda.Runtime.NODEJS_LATEST,
+      handler: 'index.handler',
+    });
+
+    const inputTransformation = InputTransformation.fromObject({
+      key: '👀',
+    });
+    const target = new LambdaFunction(lambdaFunction, { inputTransformation });
+
+    new Pipe(stack, 'MyLambdaPipe', {
+      source: new TestSource(),
+      target,
+    });
+
+    // ACT
+    const template = Template.fromStack(stack);
+
+    // ASSERT
+    template.hasResourceProperties('AWS::Pipes::Pipe', {
+      Target: {
+        'Fn::GetAtt': [
+          'MyLambdaCCE802FB',
+          'Arn',
+        ],
+      },
+      TargetParameters: {
+        InputTemplate: '{"key":"👀"}',
+        LambdaFunctionParameters: {
+          InvocationType: 'REQUEST_RESPONSE',
+        },
+      },
+    });
+  });
+
+  it('should grant lambda function invoke', () => {
+    // ARRANGE
+    const app = new App();
+    const stack = new Stack(app, 'TestLambdaFunctionStack');
+
+    const lambdaFunction = new lambda.Function(stack, 'MyLambda', {
+      code: lambda.Code.fromInline('console.log("Hello, world!")'),
+      runtime: lambda.Runtime.NODEJS_LATEST,
+      handler: 'index.handler',
+    });
+
+    const target = new LambdaFunction(lambdaFunction, {});
+
+    new Pipe(stack, 'MyLambdaPipe', {
+      source: new TestSource(),
+      target,
+    });
+
+    // ACT
+    const template = Template.fromStack(stack);
+
+    // ASSERT
+    expect(template.findResources('AWS::IAM::Role')).toMatchSnapshot();
+    template.hasResourceProperties('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: 'lambda:InvokeFunction',
+            Effect: 'Allow',
+            Resource: [
+              {
+                'Fn::GetAtt': [
+                  'MyLambdaCCE802FB',
+                  'Arn',
+                ],
+              },
+              {
+                'Fn::Join': [
+                  '',
+                  [
+                    {
+                      'Fn::GetAtt': [
+                        'MyLambdaCCE802FB',
+                        'Arn',
+                      ],
+                    },
+                    ':*',
+                  ],
+                ],
+              },
+            ],
+          },
+        ],
+        Version: '2012-10-17',
+      },
+    });
+  });
+
+  it('should work with imported function', () => {
+    // ARRANGE
+    const app = new App();
+    const stack = new Stack(app, 'TestLambdaFunctionStack');
+
+    const lambdaFunction = new lambda.Function(stack, 'MyLambda', {
```

Comment:
You don't have to define a lambda. A string of an arn should be sufficient.

Reply:
Are you sure? Shouldnt be the lambda be available for import? Because for the LambdaFunction#grantPush you need a reference to a lambda.Function.

Reply:
Because it's a unit test and Raphael is right here, you can just hardcode a fake arn and run `lambda.Function.fromFunctionArn(stack, 'MyImportedLambda', '<fake-arn>'.

Reply:
When using `lambda.Function.fromFunctionArn(stack, 'MyImportedLambda', arn)`, I got following error:
```
Cannot modify permission to lambda function. Function is either imported or $LATEST version.
If the function is imported from the same account use `fromFunctionAttributes()` API with the `sameEnvironment` flag.
If the function is imported from a different account and already has the correct permissions use `fromFunctionAttributes()` API with the `skipPermissions` flag.
```

I've used the `fromFunctionAttributes` with the `sameEnvironment` flag. However I left the test `should work with imported function` unchanged as it would not make sense to use the same import twice. An alternative would be the deletion of this test.